locator/network_topology_strategy.cc

```diff
@@ -271,11 +271,6 @@ void network_topology_strategy::validate_options(const gms::feature_service& fs)
     }
     validate_tablet_options(*this, fs, _config_options);
     for (auto& c : _config_options) {
-        if (c.first == sstring("replication_factor")) {
-            throw exceptions::configuration_exception(
-                "replication_factor is an option for simple_strategy, not "
-                "network_topology_strategy");
-        }
```

Comment:
I'd like to keep this check. And you're right, that it's not a user-facing exception anyway, so a stricter check would be good. Let's put on_internal_error here, it can be configured not to be disastrous and just throw